src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/roi_align_ref.cl

```diff
@@ -5,99 +5,103 @@
+    const INPUT1_TYPE x2 =
+        (roi_ptr[2] + (INPUT1_TYPE)OFFSET_SRC) * (INPUT1_TYPE)SPATIAL_SCALE + (INPUT1_TYPE)OFFSET_DST;
+    const INPUT1_TYPE y2 =
+        (roi_ptr[3] + (INPUT1_TYPE)OFFSET_SRC) * (INPUT1_TYPE)SPATIAL_SCALE + (INPUT1_TYPE)OFFSET_DST;
+
+
+    const INPUT1_TYPE roi_width = MAX(x2 - x1, (INPUT1_TYPE)MIN_SIZE);
+    const INPUT1_TYPE roi_height = MAX(y2 - y1, (INPUT1_TYPE)MIN_SIZE);
 
     const INPUT1_TYPE bin_width = roi_width / POOLED_WIDTH;
     const INPUT1_TYPE bin_height = roi_height / POOLED_HEIGHT;
 
-    const int sampling_ratio_x = SAMPLING_RATIO == 0 ? (int) ceil(bin_width) : SAMPLING_RATIO;
-    const int sampling_ratio_y = SAMPLING_RATIO == 0 ? (int) ceil(bin_height) : SAMPLING_RATIO;
+    const int sampling_ratio_x = SAMPLING_RATIO == 0 ? (int)ceil(bin_width) : SAMPLING_RATIO;
+    const int sampling_ratio_y = SAMPLING_RATIO == 0 ? (int)ceil(bin_height) : SAMPLING_RATIO;
+
+    const INPUT1_TYPE sample_distance_x = bin_width / (INPUT1_TYPE)sampling_ratio_x;
+    const INPUT1_TYPE sample_distance_y = bin_height / (INPUT1_TYPE)sampling_ratio_y;
+
+    const __global INPUT0_TYPE* data = src_data + INPUT0_GET_INDEX(r, c, 0, 0);
 
-    const INPUT1_TYPE sample_distance_x = bin_width / (INPUT1_TYPE) sampling_ratio_x;
-    const INPUT1_TYPE sample_distance_y = bin_height / (INPUT1_TYPE) sampling_ratio_y;
-    
-    const __global INPUT0_TYPE* data = src_data + INPUT0_OFFSET + r*INPUT0_BATCH_PITCH + INPUT0_FEATURE_PITCH*c;
     OUTPUT_TYPE pooled_value = 0;
     for (unsigned int y_sample_ind = 0; y_sample_ind < sampling_ratio_y; y_sample_ind++) {
-        INPUT1_TYPE sample_y = y1 + (INPUT1_TYPE) y * bin_height +
-            sample_distance_y * ((INPUT1_TYPE) y_sample_ind + (INPUT1_TYPE) 0.5f);
+        INPUT1_TYPE sample_y =
+            y1 + (INPUT1_TYPE)y * bin_height + sample_distance_y * ((INPUT1_TYPE)y_sample_ind + (INPUT1_TYPE)0.5f);
         for (unsigned int x_sample_ind = 0; x_sample_ind < sampling_ratio_x; x_sample_ind++) {
-            INPUT1_TYPE sample_x = x1 + (INPUT1_TYPE) x * bin_width +
-                sample_distance_x * ((INPUT1_TYPE) x_sample_ind + (INPUT1_TYPE) 0.5f);
+            INPUT1_TYPE sample_x =
+                x1 + (INPUT1_TYPE)x * bin_width + sample_distance_x * ((INPUT1_TYPE)x_sample_ind + (INPUT1_TYPE)0.5f);
             unsigned int sample_y_low = 0;
             unsigned int sample_x_low = 0;
             unsigned int sample_y_high = 0;
             unsigned int sample_x_high = 0;
-            INPUT1_TYPE weight_left = (INPUT1_TYPE) 0.f;
-            INPUT1_TYPE weight_right = (INPUT1_TYPE) 0.f;
-            INPUT1_TYPE weight_top = (INPUT1_TYPE) 0.f;
-            INPUT1_TYPE weight_bottom = (INPUT1_TYPE) 0.f;
+            INPUT1_TYPE weight_left = (INPUT1_TYPE)0.f;
+            INPUT1_TYPE weight_right = (INPUT1_TYPE)0.f;
+            INPUT1_TYPE weight_top = (INPUT1_TYPE)0.f;
```

Comment:
in general compiler should optimize it but here we can avoid casting using constants like here
https://github.com/tgubanova-lohika/openvino/blob/master/src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/experimental_detectron_prior_grid_generator_ref.cl#L6